src/unix/tty.c

```diff
@@ -120,6 +120,9 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
    * the handle queue, since it was added by uv__handle_init in uv_stream_init.
    */
 
+  if (!(flags & UV_STREAM_BLOCKING))
+    uv__nonblock(fd, 1);
+
 #if defined(__APPLE__)
```

Comment:
Yes. It's above because if UV_STREAM_BLOCKING is not set, we want the fd to be non-blocking. On OSX, if it works with kqueue() it will be added to the pollset, else we'll use select() in a thread. The call to `uv__stream_try_select` below might overwrite fd, so we were setting the wrong fd as non-blocking.